### DISCUSSION TITLE
How to stop a name with a "." being auto nested

### DISCUSSION THREAD

**Discussion by BatchCodes** [react-hook-form/react-hook-form#4712] [General]
Is there a way to stop the data being received by the function passes to `handleSubmit` having the `name`s automatically nested if they contain `.` ?

For example in the following snippet (and linked [codesandbox](https://codesandbox.io/s/react-hook-form-get-started-forked-58b6x?file=/src/index.js))
```
    <form onSubmit={handleSubmit(onSubmit)}>
      <label>Example</label>
      <input name="thing.first" defaultValue="test" ref={register} />
      <label>ExampleRequired</label>
      <input
        name="thing.second"
        ref={register({ required: true, maxLength: 10 })}
      />
      {errors.exampleRequired && <p>This field is required</p>}
      <input type="submit" />
    </form>
```
I would like the object received by the `onSubmit` function to be `{"thing.first": "first value", "thing.second": "second value"}` rather than `{thing:{first:"first value", second:"second value"}}`

Is this possible? Thank you in advance

**Comment by bluebill1049** [comment]
not possible with our current design and the support with type check (typescript), '.' is a key character for us to convert into array or object.
